### PR TITLE
Port keyless to nextra docs

### PR DESCRIPTION
**Pull request by heliuchuan**
### Checklist

- Do all Lints pass?
  - [ ] Have you ran `pnpm spellcheck`?
  - [ ] Have you ran `pnpm fmt`?
  - [ ] Have you ran `pnpm lint`?

### CODE REVIEW

File: apps/nextra/package.json [hunk 1]
```diff
@@ -10,9 +10,7 @@
     "clean": "rimraf .next .turbo",
     "debug": "NODE_OPTIONS='--inspect' next dev",
     "dev": "next -p 3030",
-    "prebuild": "node nextra-remote-filepaths/fetch.js",
     "postbuild": "next-sitemap --config next-sitemap.config.mjs; node generate-language-sitemaps.cjs; node generate-sitemap-index.cjs",
-    "predev": "pnpm prebuild",
```
Comment (gregnazario) [inline apps/nextra/package.json:15]: I don't think these are supposed to be removed

**Review by gregnazario — COMMENTED** [review]
I can review this, but we probably need to double check everything here

File: apps/nextra/package.json (hunk 1 above)
Comment (heliuchuan) [reply apps/nextra/package.json:15]: I chatted with andrew and he said is ok to remove @hariria

File: apps/nextra/public/robots.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# *
```
Comment (heliuchuan) [inline apps/nextra/public/robots.txt:1]: i'm guessing this stuff that was generated should be taken out??

File: apps/nextra/package.json (hunk 1 above)
Comment (heliuchuan) [reply apps/nextra/package.json:15]: ill take out and maybe move to separate pr

Comment (hariria) [reply apps/nextra/package.json]: Yeah its ok to remove

File: apps/nextra/pages/en/build/guides/aptos-keyless/how-keyless-works.mdx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+**Step 4**: Once the user has signed in, Google sends the dApp a signed JWT, which includes the user's `sub` identifier (e.g., `uid-123`), the application’s `aud` identifier (e.g., `"dapp-xyz"`) and the `nonce` with the EPK commitment. (This assumes that the dApp has previously registered with Google and received this `"dapp-xyz"` identifier.)
+
+**Step 5**: The dApp now has almost everything it needs to derive a keyless account for the user: the user’s identifier (`sub`) and the dApp’s identifier (`aud`). But, to preserve the privacy of the user, the dApp will use a third piece of information: a blinding factor $r$ called a **pepper**. The dApp will contact a so-called **guardian** who will deterministically derive a random $r$ from the given (`sub`, `aud`). Importantly, the guardian will only reveal $r$ to the dApp upon seeing a validly-signed JWT for the queried (`sub`, `aud`).
+
+**Step 6**: The dApp derives the address of the account as $\mathsf{addr} = H(\texttt{"uid-123"}, \texttt{"dapp-xyz"}, r)$, where $H$ is a cryptographic hash function.
```
Comment (hariria) [inline apps/nextra/pages/en/build/guides/aptos-keyless/how-keyless-works.mdx:45]: Use the steps component https://preview.aptos.dev/en/developer-platforms/contribute/components/steps

File: apps/nextra/pages/en/build/guides/aptos-keyless/how-keyless-works.mdx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+In the previous flow, we showed how a dApp can sign in a Google user and derive their privacy-preserving keyless address, with the help of a guardian.
+
+Next, we show how this dApp can obtain a zero-knowledge proof (ZKP), which will allow it to authorize transactions from this address for the user. Importantly, the transaction will hide the user’s identifying information (e.g., the `sub` field).
+
+![Keyless proof diagram](/docs/aptos-keyless/keyless-proof.png "Keyless proof diagram")
+
+**Step 1**: The dApp sends all the necessary public information (i.e., $\mathsf{epk}$, $\mathsf{GPK}$) and private information (i.e., JWT, signature $\sigma_G$ from Google, EPK blinding factor $\rho$, and pepper $r$) to the **prover service**.
+
+**Step 2**: The prover derives the user’s address $\mathsf{addr}$ and computes a zero-knowledge proof (ZKP) $\pi$ for the keyless relation $\mathcal{R}_\mathsf{keyless}$ (described below). This proof acts as a **privacy-preserving** digital certificate, and binds the user's address $\mathsf{addr}$ to the ephemeral public key $\mathsf{epk}$. The prover then sends $\pi$ to the dApp.
```
Comment (hariria) [inline apps/nextra/pages/en/build/guides/aptos-keyless/how-keyless-works.mdx:65]: Steps component

File: apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx [hunk 5]
```diff
@@ -0,0 +1,328 @@
+---
+title: "Integration Guide"
+# id: "Aptos Keyless Integration Guide"
```
Comment (hariria) [inline apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx:3]: Remove

File: apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+
+### Step 3. Client Integration Steps
+
+Below are the default steps for a client to integrate Keyless Accounts
+
+#### 1. Present the user with a "Sign In with [IdP]" button on the UI
+
+    1. In the background, we create an ephemeral key pair. Store this in local storage.
+
+        ```tsx
```
Comment (hariria) [inline apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx:52]: just `ts` is fine

File: apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+        import {EphemeralKeyPair} from '@aptos-labs/ts-sdk';
+
+        const ephemeralKeyPair = EphemeralKeyPair.generate();
+        ```
+
+    2. Save the `EphemeralKeyPair` in local storage, keyed by its `nonce`.
+
+        ```tsx
```
Comment (hariria) [inline apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx:60]: just `ts` is fine

File: apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx [hunk 8]
```diff
@@ -0,0 +1,328 @@
...
+        // This saves the EphemeralKeyPair in local storage keyed, by its nonce.
+        storeEphemeralKeyPair(ephemeralKeyPair);
+        ```
+
+<details>
+<summary>Example implementation for `storeEphemeralKeyPair`</summary>
+
+:::tip
+This implementation is an example of how to store the `EphemeralKeyPair` in local storage using the nonce as the key. Different implementations may be used according to your application's needs.
+:::
```
Comment (hariria) [inline apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx:70]: Should be a callout component

File: apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx [hunk 9]
```diff
@@ -0,0 +1,328 @@
...
+
+```typescript
```
Comment (hariria) [inline apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx:72]: ts

File: apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx [hunk 10]
```diff
@@ -0,0 +1,328 @@
...
+      return EphemeralKeyPairEncoding.decode(e);
+    return e;
+  });
+```
+
+</details>
+
+    3. Prepare the URL params of the login URL. Set the `redirect_uri` and `client_id` to your configured values with the IdP. Set the `nonce` to the nonce of the `EphemeralKeyPair` from step 1.1.
+
+        ```tsx
```
Comment (hariria) [inline apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx:160]: ts

File: apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx [hunk 11]
```diff
@@ -0,0 +1,328 @@
...
+        const redirectUri = 'https://.../login/callback'
+        const clientId = env.IDP_CLIENT_ID
+        // Get the nonce associated with ephemeralKeyPair
+        const nonce = ephemeralKeyPair.nonce
+        ```
+
+    4. Construct the login URL for the user to authenticate with the IdP. Make sure the `openid` scope is set. Other scopes such as `email` and `profile` can be set based on your app’s needs.
+
+        ```tsx
```
Comment (hariria) [inline apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx:169]: ts

File: apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx [hunk 12]
```diff
@@ -0,0 +1,328 @@
...
+        const loginUrl = `https://accounts.google.com/o/oauth2/v2/auth?response_type=id_token&scope=openid+email+profile&nonce=${nonce}&redirect_uri=${redirectUri}&client_id=${clientId}`
+        ```
+
+    5. When the user clicks the login button, redirect the user to the `loginUrl` that was created in step 1.4.
+
+#### 2. Handle the callback by parsing the token and create a Keyless account for the user
+
+    1. Once the user completes the login flow, they will be redirected to the `redirect_uri` set in step 1. The JWT will be set in the URL as a search parameter in a URL fragment, keyed by `id_token`. Extract the JWT from the `window` by doing the following:
+
+        ```tsx
```
Comment (hariria) [inline apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx:179]: ts

File: apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx [hunk 13]
```diff
@@ -0,0 +1,328 @@
...
+          return params.get('id_token');
+        };
+
+        // window.location.href = https://.../login/google/callback#id_token=...
+        const jwt = parseJWTFromURL(window.location.href)
+        ```
+
+    2. Decode the JWT and get the extract the nonce value from the payload.
+
+        ```tsx
```
Comment (hariria) [inline apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx:193]: ts

File: apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx [hunk 14]
```diff
@@ -0,0 +1,328 @@
...
+        import { jwtDecode } from 'jwt-decode';
+
+        const payload = jwtDecode<{ nonce: string }>(jwt);
+        const jwtNonce = payload.nonce
+        ```
+
+    3. Fetch the `EphemeralKeyPair` stored in step 1.2 with the decoded nonce.
+
+
+        ```tsx
```
Comment (hariria) [inline apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx:203]: ts

File: apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx [hunk 15]
```diff
@@ -0,0 +1,328 @@
...
+        const ephemeralKeyPair = getLocalEphemeralKeyPair(jwtNonce);
+        ```
+
+<details>
+<summary>Example implementation for `getLocalEphemeralKeyPair`</summary>
+
+:::tip
+This implementation is an example of how to retrieve the `EphemeralKeyPair` from local storage using the nonce as the key. Different implementations may be used according to your application's needs. It is important that you validate the expiry timestamp of the ephemeral key pair to ensure that it is still valid.
+:::
```
Comment (hariria) [inline apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx:212]: Callout

File: apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx [hunk 16]
```diff
@@ -0,0 +1,328 @@
...
+
+```typescript
```
Comment (hariria) [inline apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx:214]: ts

File: apps/nextra/pages/en/build/guides/aptos-keyless/oidc-support.mdx [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+1. Under the "App ID" section, locate your newly created App ID and click on it.
+2. Scroll down to the "Sign in with Apple" section and click on "Edit."
+3. Add the redirect URIs that your application will use for callback after authentication. The format should be https://your-app-domain.com/auth/apple/callback.
+4. Click "Save" to update the settings.
+
+**Step 5: Set Up Your OIDC Integration**
+
+1. Use an OIDC library or framework compatible with Apple's OIDC service (e.g., Passport.js for Node.js, Spring Security for Java).
+2. Configure your application to use the client ID and private key obtained from Apple during the registration process.
+3. Set up the appropriate callback URL (https://your-app-domain.com/auth/apple/callback) for handling authentication responses from Apple.
```
Comment (hariria) [inline apps/nextra/pages/en/build/guides/aptos-keyless/oidc-support.mdx:97]: Steps component

File: apps/nextra/public/robots.txt (hunk 2 above)
Comment (hariria) [reply apps/nextra/public/robots.txt:1]: Yeah

File: apps/nextra/pages/en/build/guides/aptos-keyless/how-keyless-works.mdx (hunk 4 above)
Comment (heliuchuan) [reply apps/nextra/pages/en/build/guides/aptos-keyless/how-keyless-works.mdx:65]: I think for this i prefer no step component, as these are not actual steps for the user to take.  don't want to add to the right hand nav bar

File: apps/nextra/pages/en/build/guides/aptos-keyless/introduction.mdx [hunk 18]
```diff
@@ -0,0 +1,25 @@
+---
+title: "Introduction"
```
Comment (hariria) [inline apps/nextra/pages/en/build/guides/aptos-keyless/introduction.mdx:2]: Change it to "Keyless Introduction"

File: apps/nextra/pages/en/build/guides/aptos-keyless/introduction.mdx [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+---
+
+# Introduction
```
Comment (hariria) [inline apps/nextra/pages/en/build/guides/aptos-keyless/introduction.mdx:5]: Change it to "Keyless Introduction"

File: apps/nextra/pages/en/build/guides/aptos-keyless/oidc-support.mdx [hunk 20]
```diff
@@ -0,0 +1,107 @@
+---
+title: "OIDC Support"
```
Comment (hariria) [inline apps/nextra/pages/en/build/guides/aptos-keyless/oidc-support.mdx:2]: title: "Keyless OIDC Support"

File: apps/nextra/pages/en/build/guides/aptos-keyless/other.mdx [hunk 21]
```diff
@@ -0,0 +1,56 @@
+---
+title: "Terminology and FAQ"
```
Comment (hariria) [inline apps/nextra/pages/en/build/guides/aptos-keyless/other.mdx:2]: title: "Keyless Terminology and FAQ"

File: apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx [hunk 22]
```diff
@@ -0,0 +1,327 @@
+---
+title: "Integration Guide"
```
Comment (hariria) [inline apps/nextra/pages/en/build/guides/aptos-keyless/integration-guide.mdx:2]: title: "Keyless Integration Guide"

**Review by hariria — COMMENTED** [review]
Each title is used for rendering the SEO card for the page since we use dynamic SEO cards, so I think we should have some more specificty for Keyless, the current titles are a bit ambiguous

**Review by hariria — APPROVED** [review]

**Pull request merged**